platform/features/table/src/Table.js

```diff
@@ -0,0 +1,163 @@
+        }
+
+        /**
+         * Build column definitions based on supplied telemetry metadata
+         * @param metadata Metadata describing the domains and ranges available
+         * @returns {Table} This object
+         */
+        Table.prototype.buildColumns = function(metadata) {
+            var self = this;
+
+            this.columns = [];
+
+            if (metadata) {
+                metadata.forEach(function (metadatum) {
+                    //Push domains first
+                    metadatum.domains.forEach(function (domainMetadata) {
+                        self.addColumn(new DomainColumn(domainMetadata, self.telemetryFormatter));
+                    });
+                    metadatum.ranges.forEach(function (rangeMetadata) {
+                        self.addColumn(new RangeColumn(rangeMetadata, self.telemetryFormatter));
+                    });
+                });
+            }
+            return this;
+        };
+
+        /**
+         * Add a column definition to this Table
+         * @param {RangeColumn | DomainColumn | NameColumn} column
+         * @param {Number} [index] Where the column should appear (will be
+         * affected by column filtering)
+         */
+        Table.prototype.addColumn = function (column, index) {
+            if (typeof index === 'undefined') {
+                this.columns.push(column);
+            } else {
+                this.columns.splice(index, 0, column);
+            }
+        };
+
+        /**
+         * @private
+         * @param column
+         * @returns {*|string}
+         */
+        Table.prototype.getColumnTitle = function (column) {
+                return column.getTitle();
+        };
+
+        /**
+         * Get a simple list of column titles
+         * @returns {Array} The titles of the columns
+         */
+        Table.prototype.getHeaders = function() {
+            var self = this;
+            return this.columns.map(function (column){
+                return self.getColumnTitle(column);
+            });
+        };
+
+        /**
+         * Retrieve and format values for a given telemetry datum.
+         * @param telemetryObject The object that the telemetry data is
+         * associated with
+         * @param datum The telemetry datum to retrieve values from
+         * @returns {Object} Key value pairs where the key is the column
+         * title, and the value is the formatted value from the provided datum.
+         */
+        Table.prototype.getRowValues = function(telemetryObject, datum) {
+            var self = this;
+            return this.columns.reduce(function(rowObject, column){
+                var columnTitle = self.getColumnTitle(column),
+                    columnValue = column.getValue(telemetryObject, datum);
+
+                if (columnValue !== undefined && columnValue.text === undefined){
+                    columnValue.text = '';
+                }
+                // Don't replace something with nothing.
```

Comment:
It seems that multiple columns with the same column title should be combined, and no datum should provide two different values for the same key.  How does this occur?